src/Domain/Services/CartCheckoutPageFormat.php

```diff
@@ -0,0 +1,365 @@
+
+				// Other cases coming soon - ( block | shortcode ) x ( cart | checkout ).
+			}
+		);
+	}
+
+	/**
+	 * Register hooks for admin pages.
+	 */
+	public function admin_init() {
+		add_filter(
+			'woocommerce_settings_pages',
+			function ( $pages ) {
+				$insert_after_index = array_search(
+					[
+						'type' => 'sectionend',
+						'id'   => 'checkout_process_options',
+					],
+					$pages,
+					true
+				);
+				if ( $insert_after_index ) {
+					// This includes a fudge factor to account for SSL items
+					// which don't have numeric indices.
+					// This could break if Woo core $pages content changes,
+					// or if other extensions filter/modify this index.
+					// If that happened, the items would be rendered in the
+					// wrong place in the settings page, so should only be a UX/
+					// cosmetic issue.
+					$insert_after_index = $insert_after_index + 3;
+				}
+				$insert_after_index    = min( count( $pages ), $insert_after_index );
+				$checkout_format_items = self::get_checkout_format_items();
+				array_splice( $pages, $insert_after_index, 0, $checkout_format_items );
+				return $pages;
+			}
+		);
+
+	}
+
+	/**
+	 * Generate an array of elements for displaying in `WooCommerce > Settings > Advanced` admin page.
+	 *
+	 * @return array Array of arrays, each element with various keys (e.g. title, type, id).
+	 */
+	private static function get_checkout_format_items() {
+		$block_status = self::get_cart_checkout_status();
+
+		$block_shortcode_options = [
+			'block'     => __( 'Block', 'woo-gutenberg-products-block' ),
+			'shortcode' => __( 'Shortcode', 'woo-gutenberg-products-block' ),
+			'custom'    => __( 'Custom', 'woo-gutenberg-products-block' ),
+		];
+
+		$items[] = [
+			'title' => __( 'Cart and checkout format', 'woo-gutenberg-products-block' ),
+			'type'  => 'title',
+
+			'desc'  => sprintf(
+				// Translators: %s Documentation url.
+				__( 'Choose between block or shortcode formats. <a href="%s">Learn more.</a>', 'woo-gutenberg-products-block' ),
+				'https://docs.woocommerce.com/document/cart-checkout-blocks-support-status/'
+			),
+
+			__( 'Choose between block or shortcode formats. Learn more.', 'woo-gutenberg-products-block' ),
+			'id'    => 'cart_checkout_format_options_title',
+		];
+
+		$description = 'Cart page is not set – select a page in the section above.';
+		if ( $block_status['cart_page_id'] ) {
+			$edit_url    = get_edit_post_link( $block_status['cart_page_id'] );
+			$description = sprintf(
+				// Translators: %s edit url for the page.
+				__( '<a href="%s">Edit cart page</a>', 'woo-gutenberg-products-block' ),
+				get_edit_post_link( $block_status['cart_page_id'] )
+			);
+		}
+		$items[]     = [
+			'title'   => __( 'Cart format', 'woo-gutenberg-products-block' ),
+			'desc'    => $description,
+			'type'    => 'select',
+			'id'      => 'cart_checkout_format_options_cart_format',
+			'value'   => $block_status['cart_page_format'],
+			'options' => $block_shortcode_options,
```

Comment:
What do you think about making it disabled when the page is not defined?
```PHP
			'custom_attributes' => [ 'disabled' => true ],
```

Reply:
Disabling (on page load) when the page is unset will definitely help. We still have an issue if the user changes the cart page dynamically, but that's more of an edge case. 94b41db063b44b07d23280953c006f5e3efb082c

Thanks for the tip about `custom_attributes`, I was not aware of that 😃